File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Cpp/src/SupervisorSupportHandler.cpp
#include "../include/SupervisorSupportHandler.h"
#include <iostream>

void SupervisorSupportHandler::handleRequest(const SupportRequest& request) {
    if (request.getType() == "intermediate") {
        std::cout << "SupervisorSupportHandler: Handling intermediate request.\n";
    } else if (nextHandler) {
        nextHandler->handleRequest(request);
    } else {
        std::cout << "SupervisorSupportHandler: No handler found for " << request.getType() << "\n";
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Chain of Responsibility/Java/SupervisorSupportHandler.java
public class SupervisorSupportHandler extends SupportHandler {

    @Override
    public void handleRequest(SupportRequest request) {
        if (request.getType().equalsIgnoreCase("intermediate")) {
            System.out.println("SupervisorSupportHandler: Handling intermediate request.");
        } else if (nextHandler != null) {
            nextHandler.handleRequest(request);
        } else {
            System.out.println("SupervisorSupportHandler: No handler found for " + request.getType());
        }
    }
}
